arangod/Agency/MoveShard.cpp

```diff
@@ -752,9 +754,24 @@ arangodb::Result MoveShard::abort() {
     return result;
   }
 
+  // Can now only be PENDING
   // Find the other shards in the same distributeShardsLike group:
   std::vector<Job::shard_t> shardsLikeMe =
       clones(_snapshot, _database, _collection, _shard);
+
+
+  // We can no longer abort, if any of the shards of the distributeShardsLike
+  // group has already gone from _A, B ... to B, ..., _A in Plan.
+  for (auto const& i : shardsLikeMe) {
+    auto const& cur = _snapshot.hasAsArray(
+      curColPrefix + _database + i.collection + "/" + i.shard + "/" + "servers");
+    if (cur.second && cur.first.slice()[0].copyString() == _to) {
```

Comment:
This throws if `servers` is not an array, has length 0 or the first element is not a string.